src/chromatix/utils/czt.py

```diff
@@ -0,0 +1,193 @@
+from typing import List
+
+import jax.numpy as jnp
+from chex import Array
+
+
+def _verify_axes(x, axes):
+    axes = [a + x.ndim if a < 0 else a for a in axes]
+    if any(a >= x.ndim or a < 0 for a in axes):
+        raise ValueError("axes exceeds dimensionality of input")
+    if len(set(axes)) != len(axes):
+        raise ValueError("all axes must be unique")
+    return axes
+
+
+def _verify_cztn_input(x, a, w, m, axes):
```

Comment:
This is where we need to make some big changes. Jitting doesn't allow stuff like this as it depends on the incoming values AND isn't functional. Our options are:
1) Not do it at all.
2) Use Jaxtyping: https://github.com/patrick-kidger/jaxtyping
3) Use checkify: https://docs.jax.dev/en/latest/debugging/checkify_guide.html#

I lean more towards taking this all out, and figuring out at a later stage how to do error handling nicely. I know this is good practice in numpy, but it doesn't work in Jax, unfortunately.

Reply:
I'll look into checkify for some of the simpler checks. But won't get into jaxtyping since I don't see examples of it in the codebase and that would probably warrant a longer discussion...

Reply:
I added `checkify` checks for `m` and `axis`. For checking that `a` and `w` are close to the unit circle, I couldn't see a clean way to do what with `checkify` (equality checks won't work because due to numerical errors `abs(w)` is not always exactly 1). So I've added in the docstrings that they should be on the unit circle.

Reply:
So I just checked the checkify docs and we'd need to add the transformation as well, and then handle the error. So it's a pretty complicated move. Jaxtyping does do checks during compile time, so I think I'd rather go with that. Should we just remove the check for now (commenting out is fine, with a note), and then look into jaxtyping and do this properly once and for all? WDYT?